src/homepageExperience/components/steps/arduino/WriteData.tsx

```diff
@@ -1,71 +1,107 @@
 // Libraries
-import React, {useState} from 'react'
+import React from 'react'
 
 // Components
-import {Button, ButtonGroup, ComponentColor} from '@influxdata/clockface'
+import CodeSnippet from 'src/shared/components/CodeSnippet'
 
-// Styles
-import './ArduinoSteps.scss'
+// Utils
+import {event} from 'src/cloud/utils/reporting'
 
-// Types
-type CurrentDataSelection = 'URL' | 'CSV'
-type OwnProps = {
-  bucket: string
-}
+export const WriteData = () => {
+  const codeSnippetOne = `void setup() {
+    // ... code in setup() from Initialize Client
+   
+    // Add tags to the data point
+    sensor.addTag("device", DEVICE);
+    sensor.addTag("SSID", WiFi.SSID());
+   }`
 
-export const WriteDataComponent = (props: OwnProps) => {
-  const {bucket} = props
+  const codeSnippetTwo = `void loop() {
+    // Clear fields for reusing the point. Tags will remain the same as set above.
+    sensor.clearFields();
+  
+    // Store measured value into point
+    // Report RSSI of currently connected network
+    sensor.addField("rssi", WiFi.RSSI());
+  
+    // Print what are we exactly writing
+    Serial.print("Writing: ");
+    Serial.println(sensor.toLineProtocol());
+  
+    // Check WiFi connection and reconnect if needed
+    if (wifiMulti.run() != WL_CONNECTED) {
+      Serial.println("Wifi connection lost");
+    }
+  
+    // Write point
+    if (!client.writePoint(sensor)) {
+      Serial.print("InfluxDB write failed: ");
+      Serial.println(client.getLastErrorMessage());
+    }
+  
+    Serial.println("Waiting 1 second");
+    delay(1000);
+    }`
 
-  const [currentDataSelection, setCurrentDataSelection] = useState<
-    CurrentDataSelection
-  >('URL')
+  // Events log handling
+  const logCopyCodeSnippet = () => {
+    event(`firstMile.arduinoWizard.buckets.code.copied`) // edit
+  }
 
   return (
     <>
       <h1>Write Data</h1>
-      <h2 style={{marginBottom: '8px'}}>Choose your sample data source</h2>
       <p className="small-margins">
-        We recommend choosing an option that most closely matches how you will
-        write your actual data.
+        To start writing data, append the lines of code to add tags to the Point
+        at the end of the{' '}
+        <code className="homepage-wizard--code-highlight">void setup()</code>{' '}
+        function.
+      </p>
+      <CodeSnippet
+        text={codeSnippetOne}
+        onCopy={logCopyCodeSnippet}
+        language="arduino"
+      />
+      <p>
+        Add the following{' '}
+        <code className="homepage-wizard--code-highlight">loop()</code> code
+        snippet to your sketch
+      </p>
+      <CodeSnippet
+        text={codeSnippetTwo}
+        onCopy={logCopyCodeSnippet}
+        language="arduino"
+      />
+      <p>
+        In the above code snippet, we retrive the RSSI (Received Signal Strength
+        Indicator) of your wifi connection and write it to InfluxDB using the
+        client.
+      </p>
```

Comment:
Right about here, we should add this section as present in other onboardings. 
<img width="1062" alt="Screen Shot 2022-08-17 at 2 01 26 PM" src="https://user-images.githubusercontent.com/18511823/185242417-2899ebf7-be82-4b2a-9a94-27c2cf380766.png">